src/liger_kernel/ops/cross_entropy.py

```diff
@@ -10,11 +13,14 @@ def liger_cross_entropy_kernel(
     Y_ptr,
     Y_stride,
     loss_ptr,
+    z_loss_ptr,
     loss_stride,
     n_cols,
     n_non_ignore,
     ignore_index,
     label_smoothing: tl.constexpr,
+    lse_square_scale: tl.constexpr,
```

Comment:
I'm not sure if making label_smoothing and lse_square_scale `tl.constexpr` is a correct move.
Not familiar with model training. Are these two parameters often changed in practice? I'm worried that it might cause the same issue as #146.

[Flash-attention's implementation](https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/ops/triton/cross_entropy.py) creates a new constexpr for it in `triton.heuristics` to solve branching issues. 
I wonder what the difference is between 
1) declare`label_smoothing` as a constexpr, and
2) do calculations in `triton.heuristics` then assign a value to the constexpr `HAS_SMOOTHING`

My assumption is that:
in case 1, JIT every time `label_smoothing` changes
in case 2, JIT only when  `HAS_SMOOTHING` changes because of calculations on `label_smoothing`.

If so, I will go with flash-attn's approach.